core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSupervisor.scala

```diff
@@ -26,56 +26,95 @@ import scala.util.control.NonFatal
+   *  #3 and #4 are meant to be used as a way to diagnose the behaviour of a particular workspace or submission.
+   *
+   *  Every SubmissionMonitorActor calculates #3 and sends it in the SaveCurrentWorkflowStatusCounts message. Therefore two submissions
+   *  running in the same workspace will update activeSubmissionStatusCounts to the same value once per poll interval. This is silly but fine.
+   *
+   *  The fields below are marked volatile because they are read by a separate statsd thread.
+   *  They are only written by this actor.
+   */
   @volatile
-  private var currentWorkflowStatusCounts: Map[WorkflowStatus, Int] = Map.empty
+  private var activeSubmissionStatusCounts: Map[WorkspaceName, Map[SubmissionStatus, Int]] = Map.empty
   @volatile
-  private var currentSubmissionStatusCounts: Map[SubmissionStatus, Int] = Map.empty
+  private var activeWorkflowStatusCounts: Map[UUID, Map[WorkflowStatus, Int]] = Map.empty
+  @volatile
+  private var globalSubmissionStatusCounts: Map[SubmissionStatus, Int] = Map.empty
+  @volatile
+  private var globalWorkflowStatusCounts: Map[WorkflowStatus, Int] = Map.empty
+
+  override def preStart(): Unit = {
+    super.preStart()
+
+    initGlobalJobExecGauges()
+    system.scheduler.schedule(0 seconds, submissionPollInterval, self, UpdateGlobalJobExecGauges)
+  }
 
   override def receive = {
-    case SubmissionStarted(workspaceContext, submissionId, credential) =>
-      val child = startSubmissionMonitor(workspaceContext, submissionId, credential)
+    case SubmissionStarted(workspaceName, submissionId) =>
+      val child = startSubmissionMonitor(workspaceName, submissionId, bucketCredential)
       scheduleNextCheckCurrentWorkflowStatus(child)
-      initGauge(workspaceContext, submissionId)
+      initGauge(workspaceName, submissionId)
 
-    case SaveCurrentWorkflowStatusCounts(workflowStatusCounts, submissionStatusCounts, reschedule) =>
-      this.currentWorkflowStatusCounts = workflowStatusCounts
-      this.currentSubmissionStatusCounts = submissionStatusCounts
+    case SaveCurrentWorkflowStatusCounts(workspaceName, submissionId, workflowStatusCounts, submissionStatusCounts, reschedule) =>
+      this.activeWorkflowStatusCounts += submissionId -> workflowStatusCounts
+      this.activeSubmissionStatusCounts += workspaceName -> submissionStatusCounts
       if (reschedule) {
         scheduleNextCheckCurrentWorkflowStatus(sender)
+      } else {
+        //this submission is complete; save some metrics and some memory by removing its workflow status counter gauge
+        unregisterWorkflowGauges(workspaceName, submissionId)
+
+        //if all the submissions in this workspace are in terminal statuses, we can unregister the gauge for its submission count too
+        if( submissionStatusCounts.values.sum == submissionStatusCounts.filterKeys( SubmissionStatuses.terminalStatuses contains _ ).values.sum ) {
```

Comment:
Would this suffice?
```
if (submissionStatusCounts.keySet == SubmissionStatuses.terminalStatuses.toSet)
```

Reply:
yes! thanks, i was thinking this was ridiculous as i wrote it

Reply:
Actually it turns out that terminal statuses has multiple statuses in it (Done and Aborted) so this doesn't work. But `(submissionStatusCounts.keySet -- SubmissionStatuses.terminalStatuses.toSet).isEmpty ` does.